From 731897bc85701e71bcf0e63e76d8b12675dca934 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Karel=20Ko=C4=8D=C3=AD?= <cynerd@email.cz>
Date: Sun, 29 Mar 2020 12:35:09 +0200
Subject: [PATCH] base-files: Setup LEDs on Omnia

This configures network LEDs triggers for Turris Omnia on first boot. It
looks for ieee80211 PCIe cards and if it founds it in PCI slot then it
configured appropriate wireless lan interface as trigger.
Note that this matches only in default configuration.

Original author: Michal Hrusecky <Michal@Hrusecky.net>
---
 .../mvebu/cortexa9/base-files/etc/board.d/02_network   | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/target/linux/mvebu/cortexa9/base-files/etc/board.d/02_network b/target/linux/mvebu/cortexa9/base-files/etc/board.d/02_network
index 44188c5..07251a8 100755
--- a/target/linux/mvebu/cortexa9/base-files/etc/board.d/02_network
+++ b/target/linux/mvebu/cortexa9/base-files/etc/board.d/02_network
@@ -15,6 +15,16 @@ mvebu_setup_interfaces()
 	case "$board" in
 	cznic,turris-omnia)
 		ucidef_set_interfaces_lan_wan "lan0 lan1 lan2 lan3 lan4" "eth2"
+		ucidef_set_led_netdev "wan" "wan" "omnia-led:wan" "eth2"
+		for i in $(seq 1 3); do
+			pcipath="/sys/bus/pci/devices/0000:0$i:00.0/ieee80211"
+			phyx="$(find "$pcipath" -name 'phy*' -maxdepth 1 2>/dev/null)"
+			if [ -n "$phyx" ]; then
+				ucidef_set_led_netdev "pci$i" "pci$i" "omnia-led:pci$i" "wlan${phyx#$pcipath/phy}"
+			else
+				ucidef_set_led_default "pci$i" "pci$i" "omnia-led:pci$i" "0"
+			fi
+		done
 		;;
 	linksys,wrt1200ac|\
 	linksys,wrt1900ac-v1|\
-- 
2.27.0

